### PR TITLE
L1-238: Substrate v1.6.0

### PR DESCRIPTION
**Pull request by lesniak43**
# Description

Substrate update to v1.6.0.

### CODE REVIEW

File: bin/runtime/src/lib.rs [hunk 1]
```diff
@@ -784,6 +788,12 @@ impl pallet_identity::Config for Runtime {
     type Slashed = Treasury;
     type ForceOrigin = EnsureRoot<AccountId>;
     type RegistrarOrigin = EnsureRoot<AccountId>;
+    type OffchainSignature = Signature;
+    type SigningPublicKey = <Signature as Verify>::Signer;
+    type UsernameAuthorityOrigin = EnsureRoot<AccountId>;
+    type PendingUsernameExpiration = ConstU32<{ 7 * 24 * 60 * 60 * 1000 }>;
```
Comment (kostekIV) [inline bin/runtime/src/lib.rs:794]: Name this const

File: bin/runtime/src/lib.rs [hunk 2]
```diff
@@ -154,6 +154,7 @@ impl frame_system::Config for Runtime {
     type AccountId = AccountId;
     /// The aggregated dispatch type that is available for extrinsics.
     type RuntimeCall = RuntimeCall;
+    type RuntimeTask = RuntimeTask;
```
Comment (kostekIV) [inline bin/runtime/src/lib.rs:157]: nit: add short doc

**Review by kostekIV — APPROVED** [review]

File: bin/runtime/src/lib.rs [hunk 3]
```diff
@@ -575,6 +577,7 @@ impl pallet_staking::Config for Runtime {
     type OffendingValidatorsThreshold = OffendingValidatorsThreshold;
     type VoterList = pallet_staking::UseNominatorsAndValidatorsMap<Runtime>;
     type MaxUnlockingChunks = ConstU32<16>;
+    type MaxControllersInDeprecationBatch = ConstU32<64>;
```
Comment (Marcin-Radecki) [inline bin/runtime/src/lib.rs:580]: Let's make this const 4084, which is a number unique controler-stash pairs on Mainnet
```
marcin@marcin-Latitude-3520 ~/g/polkadot-helper-scripts (main)> yarn && yarn controllers 
[...]
aleph_testnet stats:
Total records:  6998
Unique pairs:  658
Percentage still unique:  9.40 %
Unique pairs that are active:  646
Percentage of unique pairs that are active:  98.18 %
[...}
aleph_mainnet stats:
Total records:  13767
Unique pairs:  4084
Percentage still unique:  29.67 %
Unique pairs that are active:  3853
Percentage of unique pairs that are active:  94.34 %
```

**Review by Marcin-Radecki — APPROVED** [review]

File: bin/runtime/src/lib.rs (hunk 1 above)
Comment (lesniak43) [reply bin/runtime/src/lib.rs]: done

File: bin/runtime/src/lib.rs (hunk 2 above)
Comment (lesniak43) [reply bin/runtime/src/lib.rs:160]: done

File: bin/runtime/src/lib.rs (hunk 3 above)
Comment (lesniak43) [reply bin/runtime/src/lib.rs]: done

**Pull request merged**